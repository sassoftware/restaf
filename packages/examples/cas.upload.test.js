/*
 * ------------------------------------------------------------------------------------
 *   Copyright (c) SAS Institute Inc.
 *   Licensed under the Apache License, Version 2.0 (the "License");
 *   you may not use this file except in compliance with the License.
 *   You may obtain a copy of the License at
 *
 *   http://www.apache.org/licenses/LICENSE-2.0
 *
 *   Unless required by applicable law or agreed to in writing, software
 *   distributed under the License is distributed on an "AS IS" BASIS,
 *   WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
 *   See the License for the specific language governing permissions and
 *   limitations under the License.
 * ---------------------------------------------------------------------------------------
 *
 */

'use strict';
let testFunctions = require('./testFunctions');
let testInfo;
beforeAll(async () => {
	testInfo = await require('./lib/setupAll')();
});

test('upload SAS program using restaflib', async () => {
	expect.assertions();
	let l = await testFunctions.casUploadProgram2(true, testInfo);
	expect(l).toMatchSnapshot();
});


test('upload a CSV using restaflib', async () => {
	expect.assertions();
	let l = await testFunctions.casUploadCsv2(true, testInfo);
	expect(l).toMatchSnapshot();
});


test('upload a bdat using restaflib', async () => {
	expect.assertions();
	let l = await testFunctions.casUploadbdat(true, testInfo);
	expect(l).toMatchSnapshot();
});


test('upload a score hdat and score it using restaflib', async () => {
	expect.assertions();
	let l = await testFunctions.casUploadhdat(true, testInfo);
	expect(l).toMatchSnapshot();
});

test('upload astore using restaflib', async () => {
	expect.assertions();
	let l = await testFunctions.casUploadAst(true, testInfo);
	expect(l).toMatchSnapshot();
});
