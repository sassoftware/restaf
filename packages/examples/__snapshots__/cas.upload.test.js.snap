// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`append cas table 1`] = `"done"`;

exports[`upload a CSV using casUpload 1`] = `
Array [
  Array [
    1,
    "Acura",
    "MDX                    ",
    "SUV   ",
    "Asia  ",
    "All  ",
    36945,
    33337,
    3.5,
    6,
    265,
    17,
    23,
    4451,
    106,
    189,
  ],
  Array [
    2,
    "Acura",
    "RSX Type S 2dr         ",
    "Sedan ",
    "Asia  ",
    "Front",
    23820,
    21761,
    2,
    4,
    200,
    24,
    31,
    2778,
    101,
    172,
  ],
  Array [
    3,
    "Acura",
    "TSX 4dr                ",
    "Sedan ",
    "Asia  ",
    "Front",
    26990,
    24647,
    2.4,
    4,
    200,
    22,
    29,
    3230,
    105,
    183,
  ],
  Array [
    4,
    "Acura",
    "TL 4dr                 ",
    "Sedan ",
    "Asia  ",
    "Front",
    33195,
    30299,
    3.2,
    6,
    270,
    20,
    28,
    3575,
    108,
    186,
  ],
  Array [
    5,
    "Acura",
    "3.5 RL 4dr             ",
    "Sedan ",
    "Asia  ",
    "Front",
    43755,
    39014,
    3.5,
    6,
    225,
    18,
    24,
    3880,
    115,
    197,
  ],
  Array [
    6,
    "Acura",
    "3.5 RL w/Navigation 4dr",
    "Sedan ",
    "Asia  ",
    "Front",
    46100,
    41100,
    3.5,
    6,
    225,
    18,
    24,
    3893,
    115,
    197,
  ],
  Array [
    7,
    "Acura",
    "NSX coupe 2dr manual S ",
    "Sports",
    "Asia  ",
    "Rear ",
    89765,
    79978,
    3.2,
    6,
    290,
    17,
    24,
    3153,
    100,
    174,
  ],
  Array [
    8,
    "Audi ",
    "A4 1.8T 4dr            ",
    "Sedan ",
    "Europe",
    "Front",
    25940,
    23508,
    1.8,
    4,
    170,
    22,
    31,
    3252,
    104,
    179,
  ],
  Array [
    9,
    "Audi ",
    "A41.8T convertible 2dr ",
    "Sedan ",
    "Europe",
    "Front",
    35940,
    32506,
    1.8,
    4,
    170,
    23,
    30,
    3638,
    105,
    180,
  ],
  Array [
    10,
    "Audi ",
    "A4 3.0 4dr             ",
    "Sedan ",
    "Europe",
    "Front",
    31840,
    28846,
    3,
    6,
    220,
    20,
    28,
    3462,
    104,
    179,
  ],
]
`;
